server/src/main/java/org/opensearch/ingest/IngestService.java

```diff
@@ -1099,6 +1102,7 @@ void innerUpdatePipelines(IngestMetadata newIngestMetadata) {
                     processorFactories,
                     scriptService
                 );
+                IngestPipelineValidator.validateIngestPipeline(newPipeline, clusterService);
```

Comment:
No need to add the validation again in `innerUpdatePipelines`, as it is called during the applying cluster state i.e. after the cluster state was updated with IngestMetadata. 

We have added validation during pipeline creation itself, before submitting task to cluster manager, hence we it would have already validated once it reaches in cluster state applied.